flow/react-native-host-hooks.js

```diff
@@ -83,5 +83,5 @@ declare module 'UIManager' {
   ) : Promise<any>;
 }
 declare module 'View' {
-  declare var exports : typeof ReactComponent;
+  declare var exports : typeof React$Component;
```

Comment:
Can we use React.Component here instead? I thought that's the recommended alias.

Reply:
Based on [this list of React types in Flow](https://github.com/facebook/flow/blob/v0.53.1/lib/react.js), `React$Component` seems a legit type. Also, commit 3a920c6 was me running the [`flow-upgrade` utility](https://yarnpkg.com/en/package/flow-upgrade) provided by Flow, and it also inserted a bunch of `React$Component` types so... it seems okay?

Reply:
It is a legit type but it's the internal name. I don't think it's meant to be used directly. The "friendly" alias is recommended for real use.

Reply:
I might be wrong though. Maybe @calebmer can tell.

Reply:
Interesting. So I just took a closer look at [the Flow upgrade script](https://unpkg.com/flow-upgrade@1.0.3/dist/upgrades/0.53.0/ReactUtilityTypes/codemod.js) and saw this comment:

> // Update the type identifier to be `React.Element` or the
> // equivalent global.

Seems like that implies the `React$Component` type is a "global" and is safe to use? Which lines up with the usage of it that I see everywhere.

Reply:
Yes, they're global and safe to use. I didn't mean to say they're not working. It just looks ugly and if there's a "proper" way to use it, I'd prefer that. But it's not important and not blocking this diff.

Reply:
Gotcha. I believe I misunderstood what you meant by "internal name". 😄 Thanks for clarifying.

Seeing as how we were already using these types (via the `ReactComponent` and `ReactElement` aliases we declared) I think it's save to move forward with this PR and follow up later based on feedback from @calebmer!